sync/src/sys/locks/condvar.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Rust condition variable implementation used in SGX environments
+#![allow(dead_code)]
+use crate::sys::locks::Mutex;
+use mc_sgx_tstdc::Condvar as SgxCondvar;
+
+/// The condition variable backend to use with the common Rust std lib Condvar
+/// interface
+pub(crate) struct Condvar {
+    inner: SgxCondvar,
+}
+
+unsafe impl Send for Condvar {}
+unsafe impl Sync for Condvar {}
```

Comment:
Advantages/disadvantages of implementing these on `mc_sgx_tstdc::Condvar`?

Reply:
blarg, I forgot they are automatically derived, https://doc.rust-lang.org/nomicon/send-and-sync.html

Should probably do the same for Mutex and RwLock, I'll create a follow on PR to remove the unnecessary traits at this level.
Added as a task to #29